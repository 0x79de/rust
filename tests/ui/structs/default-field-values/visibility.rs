#![feature(default_field_values)]
pub mod foo {
    pub struct Alpha {
        beta: u8 = 42,
        gamma: bool = true,
    }
}

mod bar {
    fn baz() {
        let x = crate::foo::Alpha { .. };
        //~^ ERROR field `beta` of struct `Alpha` is private
        //~| ERROR field `gamma` of struct `Alpha` is private
    }
}

pub mod baz {
    pub struct S {
        x: i32 = 1,
    }
}
fn main() {
    let a = baz::S {
        .. //~ ERROR field `x` of struct `S` is private
    };
    let b = baz::S {
        x: 0, //~ ERROR field `x` of struct `S` is private
    };
}
