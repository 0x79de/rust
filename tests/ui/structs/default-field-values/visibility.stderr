error[E0451]: field `x` of struct `S` is private
  --> $DIR/visibility.rs:24:9
   |
LL |         ..
   |         ^^ field `x` is private

error[E0451]: field `x` of struct `S` is private
  --> $DIR/visibility.rs:27:9
   |
LL |         x: 0,
   |         ^^^^ private field

error[E0451]: field `beta` of struct `Alpha` is private
  --> $DIR/visibility.rs:11:37
   |
LL |         let x = crate::foo::Alpha { .. };
   |                                     ^^ field `beta` is private

error[E0451]: field `gamma` of struct `Alpha` is private
  --> $DIR/visibility.rs:11:37
   |
LL |         let x = crate::foo::Alpha { .. };
   |                                     ^^ field `gamma` is private

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0451`.
